### PR TITLE
web: when user opts in/out via nudge, acknowledge it, and vanish the notification [ch2590]

### PR DESCRIPTION
**Pull request by maiamcc**
Hello @jazzdan, @yuindustries,

Please review the following commits I made in branch maiamcc/acknowledge-opt:

d79f671082f9b2c739d80b725b368dcb63f9daec (2019-05-16 18:37:45 -0400)
add enzyme-to-json, test analyticsNudge

7d5497fecac9d32896fd75ef154d0ebcc21659a6 (2019-05-16 17:54:32 -0400)
web: store state on analyticsNudge, reflect endpoint result, disappear after 3s

Code review reminders, by giving a LGTM you attest that:

* Commits are adequately tested
* Code is easy to understand and conforms to style guides
* Incomplete code is marked with TODOs
* Code is suitably instrumented with logging and metrics

### CODE REVIEW

**Comment by maiamcc** [comment]
uhh I might have messed something up installing the node modules 😅

**Comment by maiamcc** [comment]
Indeed, I did something goofy with node modules. Should be fixed now, thanks @jazzdan!

File: web/src/AnalyticsNudge.tsx [hunk 1]
```diff
@@ -1,33 +1,75 @@
...
+// constructor for defaults?
+
+class AnalyticsNudge extends Component<
+  AnalyticsNudgeProps,
+  AnalyticsNudgeState
+> {
+  shouldShow(): boolean {
+    return (
+      this.props.needsNudge ||
+      (this.state && this.state.requestMade && !this.state.dismissed)
```
Comment (jazzdan) [inline web/src/AnalyticsNudge.tsx]: Let's set these to sane defaults in a constructor so we don't have to do these null checks.

File: web/src/AnalyticsNudge.tsx [hunk 2]
```diff
@@ -1,33 +1,75 @@
...
-      console.log("got response -->", response.status) // returns 200
+    }).then((response: Response) => {
+      this.setState({
+        responseCode: response.status,
+      })
+
+      // after 3s, dismiss the nudge
+      setTimeout(() => {
+        this.setState({ dismissed: true })
+      }, 3000)
```
Comment (jazzdan) [inline web/src/AnalyticsNudge.tsx]: give this number a name, like `const 3Seconds = 3000`?

File: web/src/AnalyticsNudge.tsx [hunk 3]
```diff
@@ -1,33 +1,75 @@
...
     })
   }
-  render() {
-    let classes = ["AnalyticsNudge"]
-    if (this.props.needsNudge) {
-      // or if  already visible...
-      classes.push("is-visible")
+
+  messageElem(): JSX.Element {
+    if (this.state && this.state.responseCode) {
```
Comment (jazzdan) [inline web/src/AnalyticsNudge.tsx]: We won't need these acrobatics if we set a default in the constructor.

File: web/src/AnalyticsNudge.tsx [hunk 4]
```diff
@@ -1,33 +1,75 @@
...
+            <a href="https://tilt.dev/contact" target="_blank">
+              Get in touch
+            </a>
+            .
+          </span>
+        )
+      }
     }
 
+    if (this.state && this.state.requestMade) {
```
Comment (jazzdan) [inline web/src/AnalyticsNudge.tsx]: Same here

Comment (maiamcc) [reply web/src/AnalyticsNudge.tsx]: lol way ahead of you bud

**Pull request merged**